File: src/ui/src/shared/types/Customer.ts
import {EntityBase} from "zavadil-ts-common";
import {User} from "./User";
import {Shop} from "./Shop";

export type CustomerBase = EntityBase & {
	userState: string;
}

export type Customer = CustomerBase & {
	shop: Shop;
	user: User;
}

export type CustomerStub = CustomerBase & {
	shopId: number;
	userId: number;
}

File: src/main/java/eu/zavadil/openmerch/data/shopCustomer/Customer.java
package eu.zavadil.openmerch.data.shopCustomer;

import eu.zavadil.openmerch.data.shop.Shop;
import eu.zavadil.openmerch.data.user.User;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "shop_customer")
public class Customer extends CustomerBase {

	@ManyToOne(optional = false)
	private Shop shop;

	@ManyToOne(optional = false)
	private User user;

}
